google/api_core/exceptions.py

```diff
@@ -22,12 +22,15 @@
 from __future__ import unicode_literals
 
 import http.client
+from google.rpc import error_details_pb2
```

Comment:
Two things:

- `grpcio` is a conditional dependency (some REST-only libraries don't need or want it).
- We still support older versions of it that don't have the `details_pb2` module (down to `grpcio 1.33.2`).

This import needs to be moved inside its own `try:` (because of older versions) below, and a default value set in the `except ImportError:` block.

`google.rpc`

Reply:
+1 - `error_details_pb2` comes from [googleapis-common-protos](https://github.com/googleapis/python-api-common-protos/tree/main/google/rpc).